### ISSUE TITLE
Button dropdown does not close when Escape key is pressed

### ISSUE THREAD

**Issue by schpet** [reactstrap/reactstrap#1634]
- components: `ButtonDropdown`
- reactstrap version `reactstrap@8.0.1`
- import method `webpack`
- react version `react@16.8.6`
- bootstrap version `bootstrap@4.3.1`

### What is happening?

dropdown is not closing on escape key

### What should be happening?

dropdown should close on escape key

> However, Bootstrap does add built-in support for most standard keyboard menu interactions, such as the ability to move through individual .dropdown-item elements using the cursor keys and close the menu with the ESC key.

https://getbootstrap.com/docs/4.3/components/dropdowns/#accessibility

### Steps to reproduce issue

1. click on a dropdown, opening it up
2. press `Esc` key

### Code

https://reactstrap.github.io/components/button-dropdown/

**Comment by TheSharpieOne** [comment]
Looks like we only close on <kbd>esc</kbd> (and perform other keyboard actions such as arrow keys) when focus is within the menu (not the button). When focus is on the button, <kbd>esc</kbd> and other keyboard actions, such as arrow keys, need to still work.

**Comment by ftonato** [comment]
Could I spend attention to this one?

**Comment by schpet** [comment]
whoa, awesome! really appreciate the fix folks 🙌🏻
